pandas/tests/indexing/test_indexing.py

```diff
@@ -741,13 +741,21 @@ def test_slice_with_zero_step_raises(self):
             s.loc[::0]
 
     def test_indexing_assignment_dict_already_exists(self):
-        df = DataFrame({"x": [1, 2, 6], "y": [2, 2, 8], "z": [-5, 0, 5]}).set_index("z")
+        index = Index([-5, 0, 5], name="z")
+        df = DataFrame({"x": [1, 2, 6], "y": [2, 2, 8]}, index=index)
         expected = df.copy()
         rhs = {"x": 9, "y": 99}
         df.loc[5] = rhs
         expected.loc[5] = [9, 99]
         tm.assert_frame_equal(df, expected)
 
+        # GH#38335 same thing, mixed dtypes
```

Comment:
was this broken too? (or just iloc as indicated)?

is `[]` ok?

Reply:
loc is broken too.  setitem is OK i think.